File: backend_java/src/main/java/br/com/fiap/chronosdtn/dto/TransactionResponse.java
package br.com.fiap.chronosdtn.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;

// Por que: Record imutável contendo as informações geradas pós processamento de banco de dados.
public record TransactionResponse(
    // Por que: ID único da transação financeira gerado pelo banco.
    Long transactionId,

    // Por que: ID do bundle DTN associado (essencial para HATEOAS e auditoria de roteador).
    Long bundleId,

    // Por que: ID do nó de origem da transação (necessário para reconstrução da chave composta do bundle).
    Long sourceNodeId,

    // Por que: ID da conta que sofreu o débito.
    Long sourceAccountId,

    // Por que: ID da conta que sofreu o crédito.
    Long destAccountId,

    // Por que: Valor transferido.
    BigDecimal amount,

    // Por que: Status consolidado da operação (PENDING, SETTLED, REJECTED).
    String settlementStatus,

    // Por que: Carimbo de data/hora do registro de liquidação.
    LocalDateTime transactionTime
) {
}

File: backend_net/ChronosDtnNetApi/DTOs/TransactionResponse.cs
// Por que: Importa o namespace System para manipulação do tipo DateTime.
using System;

// Por que: Define o namespace DTOs para organizar as classes de mapeamento de payloads de saída.
namespace ChronosDtnNetApi.DTOs;

// Por que: Record imutável contendo as informações da transação financeira persistida.
public record TransactionResponse(
    // Por que: ID único da transação financeira gerado pelo banco.
    long TransactionId,

    // Por que: ID do bundle DTN de rede que foi associado para carregar a transação.
    long BundleId,

    // Por que: ID do nó de origem da transação (necessário para HATEOAS do bundle).
    long SourceNodeId,

    // Por que: Conta financeira que sofreu o débito.
    long SourceAccountId,

    // Por que: Conta financeira que recebeu o crédito.
    long DestAccountId,

    // Por que: Valor monetário total transferido.
    decimal Amount,

    // Por que: Status de liquidação (PENDING, SETTLED, REJECTED).
    string SettlementStatus,

    // Por que: Carimbo de data/hora no qual a transação foi processada.
    DateTime TransactionTime
);
